src/Tests/Microsoft.NET.TestFramework/NuGetTransientErrorDetector.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.NET.TestFramework
+{
+    public static class NuGetTransientErrorDetector
+    {
+        private static readonly List<string> _errorSubstrings = new List<string>()
+        {
+            "A connection attempt failed because the connected party did not properly respond after a period of time",
+            "Response status code does not indicate success: 5", // match any 5xx error
```

Comment:
The other errors we've seen I think was something like: *Response status code does not indicate success: 429 ()*

Should we include a string to match on 4xx error?

Reply:
4xx means "our fault". Looks like 429 is the only error make sense to retry. I'll add 429